### PR TITLE
[ACT-3014] changed to global call instead

### PR DESCRIPTION
**Pull request by georgechieng-sc**

### CODE REVIEW

File: grpc-fixture/fixture/fixture_interceptor.go [hunk 1]
```diff
@@ -10,115 +11,141 @@ import (
...
-					}
-					if info.FullMethod == "/s12.tasks.v1.ActionsService/GetAction" {
-						receivedMessageStructure := internal.Message{
-							MessageOrigin: internal.ClientMessage,
-							RawMessage:    receivedMessage,
-							Message:       nil,
-							Timestamp:     time.Time{},
+						// recurse deeper into the tree
+						message.called = true
+						gc.ChangeState(true)
```
Comment (essentialscrum) [inline grpc-fixture/fixture/fixture_interceptor.go]: not required here

File: grpc-fixture/fixture/fixture_interceptor.go [hunk 2]
```diff
@@ -10,115 +11,141 @@ import (
...
 						}
-						receivedMessageDecoded, decodeErr := f.decoder.Decode(info.FullMethod, &receivedMessageStructure)
-						if decodeErr != nil {
-							return decodeErr
+						// found the first server message so break
+						break
+					} else {
+						// wait for a client message and then proceed based on its contents
+						var receivedMessage []byte
+						err := ss.RecvMsg(&receivedMessage)
```
Comment (essentialscrum) [inline grpc-fixture/fixture/fixture_interceptor.go]: after here back to false

**Review by essentialscrum — CHANGES_REQUESTED** [review]

File: grpc-fixture/fixture/fixture_interceptor.go (hunk 1 above)
Comment (georgechieng-sc) [reply grpc-fixture/fixture/fixture_interceptor.go]: do we still require message.called be set to true or nah?

File: grpc-fixture/fixture/fixture_interceptor.go (hunk 2 above)
Comment (georgechieng-sc) [reply grpc-fixture/fixture/fixture_interceptor.go]: would it be within the if err {...} or after it?

**Review by essentialscrum — APPROVED** [review]
Thanks @georgechieng-sc

**Pull request merged**